File: app/src/main/java/com/example/neofr/habitcounter/model/HabitDao.java
package com.example.neofr.habitcounter.model;

import android.arch.persistence.room.*;

import java.util.List;

@Dao
public interface HabitDao {
    @Query("SELECT * FROM HABIT")
    List<Habit> getAll();

    @Query("SELECT * FROM HABIT WHERE name= :name")
    Habit getByName(String name);

    @Query("SELECT * FROM HABIT WHERE id= :id")
    Habit getById(long id);

    @Insert
    void insert(Habit habit);

    @Update
    void update(Habit habit);

    @Delete
    void delete(Habit habit);
}

File: app/src/main/java/com/example/neofr/habitcounter/dao/room/HabitDAO.kt
package com.example.neofr.habitcounter.dao.room

import android.arch.persistence.room.*
import com.example.neofr.habitcounter.model.Habit


@Dao
interface HabitDAO {
    @Query("SELECT * FROM HABIT")
    fun getAll(): List<Habit>

    @Query("SELECT * FROM HABIT WHERE name= :name")
    fun getByName(name: String): Habit

    @Query("SELECT * FROM HABIT WHERE id= :id")
    fun getById(id: Long): Habit

    @Insert
    fun insert(habit: Habit)

    @Update
    fun update(habit: Habit)

    @Delete
    fun delete(habit: Habit)
}